docs/contribute/plugin/intro.md

```diff
@@ -4,34 +4,99 @@ sidebar_position: 1
 
-For the other solutions, WasmEdge provides the plug-in architecture to load the plug-in shared library for easier extending of the host functions.
+A WasmEdge plug-in is a software component that extends the functionality of the WasmEdge runtime. Currently, developers can follow the guides to implement the plug-ins in [C API](develop_plugin_c.md) (recommended) or [C++](develop_plugin_cpp.md). With the help of the WasmEdge SDKs in the supported languages, developers can load and register the host functions from the plug-in shared libraries, allowing them to seamlessly integrate the plug-ins into the WasmEdge runtime as if they were part of the core runtime.
 
-With developing the plug-ins, WasmEdge SDKs in the supported languages can load and register the host functions from the plug-in shared libraries.
+```mermaid
+graph LR
+    A((Host Application)) -- Loads --> B((Plugin Shared Library))
+    B -- Registers --> C((Wasmedge Runtime))
+```
 
-In current, developers can follow the guides to implement the plug-ins in [C API (recommended)](develop_plugin_c.md) or [C++](develop_plugin_cpp.md).
+In this diagram, the _Host Application_ represents the application or environment where the Wasmedge runtime is embedded or used. The _Plug-in Shared Library_ refers to the library containing the plug-in code and functions that extend the functionality of the Wasmedge runtime. The _Wasmedge Runtime_ represents the runtime environment that executes WebAssembly modules, including the core runtime and any registered plug-ins.
+
+## Usages of WasmEdge Plug-in
+
+WasmEdge plug-ins can be used in various scenarios, such as high-performance domain-specific languages, cloud-native JavaScript runtime, edge computing, serverless, SaaS, service mesh, and blockchain applications. Moreover, WasmEdge plug-ins can be used in microservices, providing a lightweight, secure, high-performance runtime compatible with frameworks like Dapr and Kubernetes.
+
+In addition, WasmEdge plug-ins can support serverless functions for SaaS applications, allowing users to extend and customize their experience without operating their API callback servers. They can also be embedded into innovative device applications as middleware runtime, rendering interactive content on the UI, connecting to native device drivers, and accessing specialized hardware features. WasmEdge plug-ins can manage application state, exchange dynamic data (e.g., strings and arrays) with Wasm programs, and provide native host functions to manipulate data.
+
+## Benefits of Using WasmEdge Plug-in
+
+WasmEdge plug-ins are designed to extend the functionality of the WasmEdge runtime and can be helpful for developers and end-users in several ways:
+
+- **Versatility:** WasmEdge plug-ins can be developed in multiple programming languages that can compile to WebAssembly, allowing developers to write plug-ins in the most comfortable language.
```

Comment:
The plug-ins are NOT compiled into WASM. We only provides the C/Rust etc. SDKs to implement plug-ins.